core/vpn/src/network.rs

```diff
@@ -594,22 +615,34 @@ struct InternalConnection {
     pub ingress_tx: mpsc::Sender<Vec<u8>>,
 }
 
-async fn vpn_ingress_handler(rx: IngressReceiver, addr: Addr<Vpn>) {
+async fn vpn_ingress_handler(rx: IngressReceiver, addr: Addr<Vpn>, vpn_id: String) {
     let mut rx = UnboundedReceiverStream::new(rx);
     while let Some(event) = rx.next().await {
-        addr.do_send(Ingress { event });
+        if let Err(e) = addr.send(Ingress { event }).await {
+            log::error!(
+                "[vpn: {}] ingress event handler returned error: {}",
+                vpn_id,
+                e
+            );
```

Comment:
I am not sure if we want to `Disconnect` at this error. I think that even `Shutdown` would be nice when packet wasn't delivered